File: api/report/v1/report.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

// GenerateReq 生成报告请求
type GenerateReq struct {
	g.Meta    `path:"/report/generate" method:"post" tags:"Report" summary:"生成报告"`
	Type      string `json:"type" v:"required" dc:"报告类型: daily/weekly/monthly"`
	Title     string `json:"title" v:"required" dc:"报告标题"`
	StartTime string `json:"start_time" v:"required" dc:"开始时间"`
	EndTime   string `json:"end_time" v:"required" dc:"结束时间"`
}

// GenerateRes 生成报告响应
type GenerateRes struct {
	ReportID uint   `json:"report_id"`
	Title    string `json:"title"`
	Summary  string `json:"summary"`
	Status   string `json:"status"`
}

// GetReq 获取报告请求
type GetReq struct {
	g.Meta `path:"/report/:id" method:"get" tags:"Report" summary:"获取报告"`
	ID     uint `json:"id" v:"required" dc:"报告ID"`
}

// GetRes 获取报告响应
type GetRes struct {
	ID            uint   `json:"id"`
	Title         string `json:"title"`
	Type          string `json:"type"`
	Status        string `json:"status"`
	Content       string `json:"content"`
	Summary       string `json:"summary"`
	EventCount    int    `json:"event_count"`
	CriticalCount int    `json:"critical_count"`
	HighCount     int    `json:"high_count"`
	StartTime     string `json:"start_time"`
	EndTime       string `json:"end_time"`
	GeneratedBy   string `json:"generated_by"`
	ErrorMsg      string `json:"error_msg"`
	CreatedAt     string `json:"created_at"`
}

// ListReq 获取报告列表请求
type ListReq struct {
	g.Meta   `path:"/report" method:"get" tags:"Report" summary:"获取报告列表"`
	Page     int    `json:"page" d:"1" dc:"页码"`
	PageSize int    `json:"page_size" d:"10" dc:"每页数量"`
	Type     string `json:"type" dc:"报告类型"`
}

// ListRes 获取报告列表响应
type ListRes struct {
	List  []ReportItem `json:"list"`
	Total int64        `json:"total"`
}

// ReportItem 报告列表项
type ReportItem struct {
	ID            uint   `json:"id"`
	Title         string `json:"title"`
	Type          string `json:"type"`
	Status        string `json:"status"`
	Summary       string `json:"summary"`
	EventCount    int    `json:"event_count"`
	CriticalCount int    `json:"critical_count"`
	HighCount     int    `json:"high_count"`
	GeneratedBy   string `json:"generated_by"`
	ErrorMsg      string `json:"error_msg"`
	CreatedAt     string `json:"created_at"`
}

// DeleteReq 删除报告请求
type DeleteReq struct {
	g.Meta `path:"/report/:id" method:"delete" tags:"Report" summary:"删除报告"`
	ID     uint `json:"id" v:"required" dc:"报告ID"`
}

// DeleteRes 删除报告响应
type DeleteRes struct {
	Success bool `json:"success"`
}

// TemplateCreateReq 创建模板请求
type TemplateCreateReq struct {
	g.Meta      `path:"/report/template" method:"post" tags:"ReportTemplate" summary:"创建模板"`
	Name        string `json:"name" v:"required" dc:"模板名称"`
	Description string `json:"description" dc:"模板描述"`
	Type        string `json:"type" v:"required" dc:"报告类型"`
	Content     string `json:"content" v:"required" dc:"模板内容"`
	IsDefault   bool   `json:"is_default" dc:"是否默认"`
}

// TemplateCreateRes 创建模板响应
type TemplateCreateRes struct {
	ID uint `json:"id"`
}

// TemplateListReq 获取模板列表请求
type TemplateListReq struct {
	g.Meta `path:"/report/template" method:"get" tags:"ReportTemplate" summary:"获取模板列表"`
	Type   string `json:"type" dc:"报告类型"`
}

// TemplateListRes 获取模板列表响应
type TemplateListRes struct {
	List []TemplateItem `json:"list"`
}

// TemplateItem 模板列表项
type TemplateItem struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Type        string `json:"type"`
	IsDefault   bool   `json:"is_default"`
}

// ExportReq 导出报告请求
type ExportReq struct {
	g.Meta `path:"/report/:id/export" method:"get" tags:"Report" summary:"导出报告"`
	ID     uint   `json:"id" v:"required" dc:"报告ID"`
	Format string `json:"format" d:"markdown" dc:"导出格式: markdown/html/json"`
}

// ExportRes 导出报告响应
type ExportRes struct {
	Content  string `json:"content"`
	Filename string `json:"filename"`
}

// TemplateGetReq 获取模板详情请求
type TemplateGetReq struct {
	g.Meta `path:"/report/template/{id}" method:"get" tags:"ReportTemplate" summary:"获取模板详情"`
	ID     uint `json:"id" in:"path" v:"required" dc:"模板ID"`
}

// TemplateGetRes 获取模板详情响应
type TemplateGetRes struct {
	*TemplateItem
	Content string `json:"content"`
}

// TemplateUpdateReq 更新模板请求
type TemplateUpdateReq struct {
	g.Meta      `path:"/report/template/{id}" method:"put" tags:"ReportTemplate" summary:"更新模板"`
	ID          uint   `json:"id" in:"path" v:"required" dc:"模板ID"`
	Name        string `json:"name" dc:"模板名称"`
	Description string `json:"description" dc:"模板描述"`
	Content     string `json:"content" dc:"模板内容"`
	IsDefault   bool   `json:"is_default" dc:"是否默认"`
}

// TemplateUpdateRes 更新模板响应
type TemplateUpdateRes struct{}

// TemplateDeleteReq 删除模板请求
type TemplateDeleteReq struct {
	g.Meta `path:"/report/template/{id}" method:"delete" tags:"ReportTemplate" summary:"删除模板"`
	ID     uint `json:"id" in:"path" v:"required" dc:"模板ID"`
}

// TemplateDeleteRes 删除模板响应
type TemplateDeleteRes struct{}

File: web/src/services/report.ts
import api from './api'
import {
  Report,
  ReportTemplate,
  GenerateReportRequest,
  CreateTemplateRequest,
  ExportFormat,
  ApiResponse,
  PaginatedResponse,
} from '@/types'

// 后端返回的报告列表响应格式
interface ReportListResponse {
  list: Array<{
    id: number
    title: string
    type: string
    status: string
    created_at: string
  }>
  total: number
}

export const reportService = {
  // 生成报告
  async generate(data: GenerateReportRequest): Promise<Report> {
    const res = await api.post<ApiResponse<{
      report_id: number
      title: string
      summary: string
      status: string
    }>>('/report/generate', data)
    const d = res.data.data
    return {
      id: d.report_id,
      title: d.title,
      type: data.type,
      status: d.status as Report['status'],
      start_time: data.start_time || '',
      end_time: data.end_time || '',
      content: '',
      summary: d.summary,
      event_ids: '',
      subscription_ids: '',
      event_count: 0,
      critical_count: 0,
      high_count: 0,
      generated_by: 'manual',
      created_at: new Date().toISOString(),
      updated_at: new Date().toISOString(),
    }
  },

  // 获取报告列表
  async list(page = 1, pageSize = 20): Promise<PaginatedResponse<Report>> {
    const res = await api.get<ApiResponse<ReportListResponse>>('/report', {
      params: { page, page_size: pageSize },
    })
    const data = res.data.data

    const list: Report[] = (data.list || []).map(item => ({
      id: item.id,
      title: item.title,
      type: item.type as Report['type'],
      status: item.status as Report['status'],
      start_time: '',
      end_time: '',
      content: '',
      event_ids: '',
      subscription_ids: '',
      event_count: 0,
      critical_count: 0,
      high_count: 0,
      generated_by: 'manual',
      created_at: item.created_at,
      updated_at: item.created_at,
    }))

    return {
      list,
      total: data.total || 0,
      page,
      size: pageSize,
    }
  },

  // 获取单个报告
  async get(id: number): Promise<Report> {
    const res = await api.get<ApiResponse<{
      id: number
      title: string
      type: string
      status: string
      content: string
      summary: string
      created_at: string
    }>>(`/report/${id}`)
    const d = res.data.data
    return {
      id: d.id,
      title: d.title,
      type: d.type as Report['type'],
      status: d.status as Report['status'],
      start_time: '',
      end_time: '',
      content: d.content,
      summary: d.summary,
      event_ids: '',
      subscription_ids: '',
      event_count: 0,
      critical_count: 0,
      high_count: 0,
      generated_by: 'manual',
      created_at: d.created_at,
      updated_at: d.created_at,
    }
  },

  // 删除报告
  async delete(id: number): Promise<void> {
    await api.delete(`/report/${id}`)
  },

  // 导出报告
  async export(id: number, format: ExportFormat): Promise<Blob> {
    const res = await api.get<ApiResponse<{ content: string; filename: string }>>(`/report/${id}/export`, {
      params: { format },
    })
    const { content } = res.data.data
    // 将内容转换为 Blob
    const mimeTypes: Record<string, string> = {
      markdown: 'text/markdown',
      html: 'text/html',
      json: 'application/json',
    }
    return new Blob([content], { type: mimeTypes[format] || 'text/plain' })
  },

  // 获取模板列表
  async listTemplates(): Promise<ReportTemplate[]> {
    const res = await api.get<ApiResponse<{ list: Array<{
      id: number
      name: string
      description: string
      type: string
      is_default: boolean
    }> }>>('/report/template')
    return (res.data.data.list || []).map(item => ({
      id: item.id,
      name: item.name,
      description: item.description,
      type: item.type as ReportTemplate['type'],
      content: '',
      is_default: item.is_default,
      created_at: '',
      updated_at: '',
    }))
  },

  // 创建模板
  async createTemplate(data: CreateTemplateRequest): Promise<ReportTemplate> {
    const res = await api.post<ApiResponse<{ id: number }>>('/report/template', data)
    return {
      id: res.data.data.id,
      name: data.name,
      description: data.description || '',
      type: data.type,
      content: data.content,
      is_default: data.is_default || false,
      created_at: new Date().toISOString(),
      updated_at: new Date().toISOString(),
    }
  },

  // 获取单个模板
  async getTemplate(id: number): Promise<ReportTemplate> {
    const res = await api.get<ApiResponse<ReportTemplate>>(`/report/template/${id}`)
    return res.data.data
  },

  // 更新模板
  async updateTemplate(id: number, data: Partial<CreateTemplateRequest>): Promise<void> {
    await api.put(`/report/template/${id}`, data)
  },

  // 删除模板
  async deleteTemplate(id: number): Promise<void> {
    await api.delete(`/report/template/${id}`)
  },
}
